### ISSUE TITLE
Explicit type-hinting httpx.Response should yield response object

### ISSUE THREAD

**Issue by Apxdono** [floydya/declarativex#36]
## Description
TL;DR This is a really good thing to have, as type-hinting is preserved.

While not specified in the documentation it's technically possible to explicitly type-hint client methods with `httpx.Request`.
When method is type-hinted with `httpx.Request`, Pydantic will raise `pydantic.errors.PydanticSchemaGenerationError: Unable to generate pydantic-core schema for <class 'httpx.Response'>`. 

## Steps to Reproduce
```python
import httpx
from declarativex import BaseClient, http


class DummyUsersClient(BaseClient):
    # pylint: disable=unused-argument
    base_url = "https://dummyjson.com"

    @http("GET", "/users/{user_id}")
    def get_user(self, user_id: int, ) -> httpx.Response:
        ...


if __name__ == "__main__":
    response = DummyUsersClient().get_user(1)
    print(response)

```

## Expected Behavior
Behavior should be the same as if you omit type-hinting. This should be a trivial thing to add, as absence of type-hint is already [checked here](https://github.com/floydya/declarativex/blob/60920989a593212f8071111aaf0da07a228379b0/src/declarativex/models.py#L115).

## Actual Behavior
```
(omitted)
.../site-packages/pydantic/_internal/_generate_schema.py", line 397, in _unknown_type_schema
    raise PydanticSchemaGenerationError(
pydantic.errors.PydanticSchemaGenerationError: Unable to generate pydantic-core schema for <class 'httpx.Response'>. Set `arbitrary_types_allowed=True` in the model_config to ignore this error or implement `__get_pydantic_core_schema__` on your type to fully support it.

If you got this error by calling handler(<some type>) within `__get_pydantic_core_schema__` then you likely need to call `handler.generate_schema(<some type>)` since we do not call `__get_pydantic_core_schema__` on `<some type>` otherwise to avoid infinite recursion.

For further information visit https://errors.pydantic.dev/2.6/u/schema-for-unknown-type
```
## Environment
- Python version: 3.11.7
- DeclarativeX version: 1.6.5
